### ISSUE TITLE
C++ API Implementation of new datatype query function

### ISSUE THREAD

**Issue by timpokorny** [openlvc/portico#239]
Summary
----
Add support to the 1516e C++ interface for the new datatype solicitation API added in #209 

Acceptance Criteria
----
Once complete, Portico shall:

- [x] Have a C++ interface for the class hierarchy of datatypes added in #209 (see the diagram above)
- [x] Mimic the `RTIambassadorEx` interface structure and extended calls added in [RTIambassadorEx](https://github.com/openlvc/portico/blob/master/codebase/src/java/portico/hla/rti1516e/extensions/RTIambassadorEx.java)
- [x] Have C++ unit test suite to fetch data types from RTIambassador and verify their values (full test suite as in Java side is not required)

**Comment by bShibby** [comment]
I am at the second stage of this task now. I will have fix some things as I move forward into other areas but have progressed. Unit testing is underway. I updated the app example to be more ieee1516e focused. Once that is merged into master I will be able to update it again and include the new CPP get class attributes.

I have set up a dummy getFom function in the proxy class that passes back a string. The string currently is a hardcoded XML string. I will chat with Michael about what exactly we want to pass across laterz.
